docs/object.js

```diff
@@ -91,10 +91,10 @@ class Object {
     removeAllListeners() { }
 
     /**
-     * Returns a unique identifier. For synced objects, the uniqueness is only guaranteed when the object has been
+     * Returns a unique and global key. For synced objects, the uniqueness is only guaranteed when the object has been
      * synchronized.
-     * @see {@link Realm#objectForObjectId objectForObjectId}
+     * @see {@link Realm#objectForGlobalKey objectForGlobalKey}
      * @since 4.0.0
      */
-    objectId() { }
+    globalKey() { }
```

Comment:
Maybe it is worth thinking a bit about the naming of this. See https://mongodb.slack.com/archives/CKW2HG3M0/p1576660133033600

Especially when it comes to communicating it to the outside.
From my understanding, `GlobalKey` is the sync variant that isn't used by non-synced Realms. In those cases, only `ObjKey` exists.

I agree that that using `objectId` should probably be avoided no matter what to avoid conflict with MongoDB and maybe `GlobalKey` is the word we want to expose to the outside. Or maybe we want to come up with another name altogether in order to not cause internal confusion as well.

- ObjectId (taken by Mongo)... Would have been much more accurate if it used `DocumentId` 😄 
- RealmId (new name returning either internal GlobalKey/ObjKey depending on context).
- GlobalKey (overloading the internal term GlobalKey as it sometimes will return a ObjKey)`
- ObjectKey (new name, although very close to `ObjKey` and `ObjectId`)
- Maybe something else?